### ISSUE TITLE
Directory Selection Cleared on Restart using Linux for Chromebook

### ISSUE THREAD

**Issue by ecc521** [kzahel/web-server-chrome#192]
Typically, this web server will remember the directory being served, and when restarted, will begin serving that directory again. If the directory does not exist, the previous selection will be cleared.

When using this web server to server files inside of Linux for Chromebooks, however, this strategy fails. Opening the server before Linux has opened will result in the previous directory selection being cleared (as the Linux files aren't yet available).

Given that this extension is perhaps the only option for locally serving files on chromebooks (the Linux terminal has it's own 127.0.0.1, and crash doesn't let you do anything significant), it would seem like we should handle this behavior. (Perhaps check if the directory exists every second, and give a warning that the directory doesn't exist?)

**Comment by kzahel** [comment]
You're referring to if you select a crostini folder? "Linux Files" in the files pane?

**Comment by ecc521** [comment]
Correct. If you select a cronsiti folder, and restart, the server remembers
the folder you selected (as it should)... but if the server opens before
cronsiti does, it errors, and clears your selection.

**Comment by kzahel** [comment]
That's annoying. I don't think I can address it though.

**Comment by ecc521** [comment]
Not sure - is there a reason the webserver needs to clear the selection if it doesn't exist? Could it simply ask for a new selection (without clearing the invalid one), while polling every few seconds to see if the directory appeared (and the selection became valid)?

Probably only needs to run for a minute or so due to startup. 

At the moment, I don't care a  whole lot about that issue, but I'm not sure it should be closed.

**Comment by kzahel** [comment]
Sure, sorry I went on a close spree yesterday. Thanks for commenting.
I'm not sure I'll do any more updates to this project though since chrome apps are deprecated and probably in a year or two chrome apps will just disappear. (not actually certain if they'll continue to exist for chromeOs at least for a number of years?)

**Comment by ecc521** [comment]
I can certainly understand closing sprees, so no problem there. I no longer
have a significant interest in ChromeOS at the moment, as everything is at
home now, and if chrome apps are being killed, I'm not sure there's any
reason to fix this issue (that said, I still think it should stay open -
it's a known, fixable, bug)

If things do change, I might take a look at this again. As is, I'd say it's
a 'wontfix' if anything.

**Comment by kzahel** [comment]
The original Chrome App is no longer supported. The app has been rebuilt from scratch — see https://ok200.app for the new versions and to sign up for updates.
